### PR TITLE
Get eslint working with GitHub actions

### PR DESCRIPTION
**Pull request by HerbCaudill**
- Makes eslint happy either by fixing code or disabling rules (locally or globally)
- Uses a single config file and lint command at the monorepo root
- Updates the lint workflow to work with pnpm rather than yarn

### CODE REVIEW

File: packages/automerge-repo/package.json [hunk 1]
```diff
@@ -9,11 +9,10 @@
   "main": "dist/index.js",
   "scripts": {
     "build": "tsc",
-    "lint": "eslint --ext .ts src",
     "watch": "npm-watch build",
     "test:coverage": "c8 --reporter=lcov --reporter=html --reporter=text pnpm test",
     "test": "vitest",
-    "test:watch": "npm-watch test",
+    "test:waµtch": "npm-watch test",
```
Comment (pvh) [inline packages/automerge-repo/package.json:15]: what did the tiny cow say? µ

File: packages/automerge-repo/src/synchronizer/DocSynchronizer.ts [hunk 2]
```diff
@@ -330,10 +334,10 @@ export class DocSynchronizer extends Synchronizer {
     }
 
     this.#processAllPendingSyncMessages()
-    this.#processSyncMessage(message, new Date())
```
Comment (pvh) [inline packages/automerge-repo/src/synchronizer/DocSynchronizer.ts:333]: ?

**Review by pvh — COMMENTED** [review]
Couple of little things -- one is just a typo and the other I'm less clear on. Why did you remove the date argument from the processSyncMessage? Was it unused? Let's ping @alexjg or @paulsonnentag to see if they know what's up there.

File: packages/automerge-repo/package.json (hunk 1 above)
Comment (HerbCaudill) [reply packages/automerge-repo/package.json:15]: 😳

**Comment by HerbCaudill** [comment]
Yeah, eslint flagged the date parameter as unused.

**Pull request merged**